cpp/dolfinx/fem/assemble_vector_impl.h

```diff
@@ -194,36 +204,50 @@ void _lift_bc_exterior_facets(
     std::span<T> b, mdspan2_t x_dofmap,
     std::span<const scalar_value_type_t<T>> x, FEkernel<T> auto kernel,
     std::span<const std::int32_t> facets,
-    fem::DofTransformKernel<T> auto pre_dof_transform, mdspan2_t dofmap0,
-    int bs0, fem::DofTransformKernel<T> auto post_dof_transpose,
-    mdspan2_t dofmap1, int bs1, std::span<const T> constants,
+    std::tuple<mdspan2_t, int, std::span<const std::int32_t>> dofmap0,
```

Comment:
I am still not comfortable with us using tuples for these input arguments. We do not use it for any other variables, even if they are tied together. @garth-wells  what do you think?
It doesn't hurt readability and documentation to have three input arguments per direction.

Reply:
It's just to highlight which cells index which dofmap, but I agree, we should probably be consistent

Reply:
I think the change is in the right direction - it reigns in the number of arguments, grouping together related objects. Otherwise the list of arguments gets too long and this can become error prone.

We have adopted this approach in other places, e.g. to collect together the arguments that define a finite element when creating a function spaces (Python).